### PR TITLE
(650b) Complete referral page

### PR DESCRIPTION
**Pull request by jsrobertson**
## Context

This page will be redirected to from an upcoming submit referral endpoint.

## Changes in this PR

- Updated `Referral` type to include `status`
- Added `feedbackUrl` as nunjucks global to be used in multiple places
- Add complete page template
- Added complete method to referral controller
- Added `/complete` path for referral journey

### Before
N/A

### After
localhost_3000_referrals_c11fab18-dc8d-420c-9c82-d0edd373732d_complete

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: server/testutils/factories/referral.ts [hunk 1]
```diff
@@ -9,4 +9,10 @@ export default Factory.define<Referral>(() => ({
   offeringId: faker.string.uuid(),
   prisonNumber: faker.string.alphanumeric({ length: 7 }),
   referrerId: faker.string.numeric({ length: 6 }),
+  status: faker.helpers.arrayElement([
```
Comment (jsrobertson) [inline server/testutils/factories/referral.ts:12]: Could default to one of the values instead if we think that's better.

File: server/controllers/refer/referralsController.ts [hunk 2]
```diff
@@ -57,6 +57,26 @@ export default class ReferralsController {
...
 
+  complete(): TypedRequestHandler<Request, Response> {
+    return async (req: Request, res: Response) => {
+      TypeUtils.assertHasUser(req)
+
+      const { referralId } = req.params
+
+      const referral = await this.referralService.getReferral(req.user.token, referralId)
+
+      if (referral.status !== 'referral_submitted') {
```
Comment (jsrobertson) [inline server/controllers/refer/referralsController.ts:68]: Can either do this or redirect to task list

Comment (Gweaton) [reply server/controllers/refer/referralsController.ts:68]: I think a 400 here makes sense - it's unlikely they'll get here in a usual user journey so throwing an error feels appropriate.

File: server/testutils/factories/referral.ts (hunk 1 above)
Comment (Gweaton) [reply server/testutils/factories/referral.ts:12]: I think the range of options is ok here, we'll likely be explicitly setting it in tests if we're testing anything related to this field so the default value wouldn't get used anyway. I think it's good to have a range.

**Review by Gweaton — APPROVED** [review]
Looks good to me. Thanks for the atomic commits, made it really easy to review

**Pull request merged**